### PR TITLE
Fixing an issue with dates being fed to openedge databases

### PR DESCRIPTION
**Pull request by waterfoul**
If there's a better way to do this please let me know

### CODE REVIEW

File: django_pyodbc/base.py [hunk 1]
```diff
@@ -279,7 +279,6 @@ def _get_connection_string(self):
         if 'extra_params' in options:
             cstr_parts.append(options['extra_params'])
         connectionstring = ';'.join(cstr_parts)
-        print(connectionstring)
```
Comment (dlo) [inline django_pyodbc/base.py]: 👍

**Comment by dlo** [comment]
We should probably use dateutil for this since it'll cover all sorts of different cases for us. It comes with a great date parser and it's not a large dependency.

See https://dateutil.readthedocs.io/en/stable/examples.html#parse-examples for some examples.

File: django_pyodbc/compiler.py [hunk 2]
```diff
@@ -86,6 +88,23 @@ def __init__(self,*args,**kwargs):
...
                 right_sql_quote=self.connection.ops.right_sql_quote))
 
+    def compile(self, node, select_format=False):
+        retVal = super(SQLCompiler, self).compile(node, select_format)
+        if self.connection.ops.is_openedge and type(node) is where.WhereNode:
+            idx = 0
+            for val in retVal[1]:
+                if type(val) == str:
+                    match = _re_date.match(val)
+                    if match:
```
Comment (waterfoul) [inline django_pyodbc/compiler.py]: This line needs to use a regular expression in order to keep false positives low

File: django_pyodbc/compiler.py [hunk 3]
```diff
@@ -86,6 +88,23 @@ def __init__(self,*args,**kwargs):
...
+                        if type(node.children[idx].rhs) == date:
+                            retVal[1][idx] = node.children[idx].rhs
+                        elif type(node.children[idx].lhs) == date:
+                            retVal[1][idx] = node.children[idx].lhs
+                        else:
+                            retval[1][idx] = date(match.group(1), match.group(2), match.group(3))
```
Comment (waterfoul) [inline django_pyodbc/compiler.py]: This case is an edge case (more than likely one of the two above will work) so it seems silly to load a library to parse here. You could argue that it should be removed to reduce false positives and only trigger on date objects

File: django_pyodbc/compiler.py (hunk 2 above)
Comment (dlo) [inline django_pyodbc/compiler.py]: Let's just make a note that if we want to support any other formats, that we might want to consider delegating this to another library. After that, looks good!

File: django_pyodbc/compiler.py (hunk 3 above)
Comment (dlo) [inline django_pyodbc/compiler.py]: Makes sense, thanks for clarifying.

**Comment by waterfoul** [comment]
How's that

**Comment by dlo** [comment]
Perfect :)

**Comment by waterfoul** [comment]
Any chance you can publish this sometime today?

**Comment by dlo** [comment]
Just pushed.

**Comment by waterfoul** [comment]
Thanks!

**Pull request merged**